packages/bitcore-node/src/services/p2p.ts

```diff
@@ -279,10 +285,64 @@ export class P2pService {
     }
     logger.info(`${chain}:${network} up to date.`);
     this.syncing = false;
-    StateModel.collection.findOneAndUpdate({}, {$addToSet: { initialSyncComplete: `${chain}:${network}`}}, { upsert: true});
+    StateModel.collection.findOneAndUpdate(
+      {},
+      { $addToSet: { initialSyncComplete: `${chain}:${network}` } },
+      { upsert: true }
+    );
     return true;
   }
 
+  async createFork(newChain: string, parentChain: string, forkHeight: number) {
+    const blockCursor = BlockModel.collection.find({ parentChain, height: { $lt: forkHeight } }, { batchSize: 100 });
+    while (blockCursor.hasNext()) {
+      const block = await blockCursor.next();
+      const newBlock = Object.assign({}, block, { chain: newChain });
+      const mongoOperations = new Array();
+      mongoOperations.push(BlockModel.collection.insert(newBlock));
```

Comment:
can do inserts on all these as long as block processed doesnt get written until the 3 steps are done for each block